linalg/vector.cpp

```diff
@@ -1165,13 +1165,13 @@ double Vector::operator*(const Vector &v) const
    if (Device::Allows(Backend::DEBUG_DEVICE))
    {
       const int N = size;
-      auto v_data = v.Read();
-      auto m_data = Read();
+      auto vv_data = v.Read();
```

Comment:
I am OK with `i`->`ii` in loops, but `v_data`->`vv_data` looks weird to me. I personally will prefer `v_data_` or `local_v_data` instead.

Reply:
Fixed